aten/src/ATen/native/sparse/SparseCsrTensorMath.cpp

```diff
@@ -1056,7 +1065,14 @@ Tensor reduce_sparse_csr_dim0_cpu_template(const Tensor& sparse, ReductionOp rop
                             for (int64_t i=0; i<numel; i++) {
                               index_t col = columns_map_ptr[i];
                               scalar_t val = values_ptr[i];
-                              new_values_ptr[col] = rop(new_values_ptr[col], val);
+                              new_values_acc_ptr[col] = rop(new_values_acc_ptr[col], static_cast<opmath_t>(val));
+                            }
+                            for (int64_t i = 0; i < nnz; i++) {
+                              if (need_acc) {
+                                new_values_ptr[i] = static_cast<scalar_t>(new_values_acc_ptr[i]);
+                              } else {
+                                new_values_ptr[i] = new_values_acc_ptr[i];
+                              }
                             }
                           });
```

Comment:
Optimization nit continued:
```suggestion
                              new_values_acc_ptr[col] = rop(new_values_acc_ptr[col], static_cast<opmath_t>(val));
                            }
                          });
  if (need_acc) {
    new_values.copy_(new_values_acc);
  }
```